### DISCUSSION TITLE
When can c# support loop objects? like javascript.

### DISCUSSION THREAD

**Discussion by yun738243875** [dotnet/csharplang#2552] [Language Ideas] [answered]
class Person{
public int ID { get; set; }
public string Name { get; set; }
}
var person = new Person() { ID = 1, Name = "test"};
foreach(var property in person)
{
Console.WriteLine("property: " + property + ", value: "+ person[property]);
//may be output is:
//property: ID, value: 1
//property: Name, value: test
}

you can see in this url https://github.com/dotnet/coreclr/issues/24677

**Comment by YairHalberstadt** [comment, marked as answer]
I see absolutely nothing wrong with Ben Adams suggested extension method at https://github.com/dotnet/coreclr/issues/24677#issuecomment-494341978. Why do you feel this requires language support?

**Comment by VanKrock** [comment]
You can to use reflection
``` csharp
    class Program
    {
        static void Main(string[] args)
        {
            var person = new Person() { ID = 1, Name = "test" };
            foreach (var property in person.GetType().GetProperties())
            {
                Console.WriteLine("property: " + property.Name + ", value: " + property.GetValue(person));
            }
        }
    }

    class Person
    {
        public int ID { get; set; }
        public string Name { get; set; }
    }
```
